.github/workflows/integration.yml

```diff
@@ -1,16 +1,45 @@
 name: Integration Test
-on: [push]
+on: [push , pull_request]
 jobs:
   build:
     runs-on: ubuntu-latest
     steps:
       - name: Check Out Repository
         id: checkout_repository
         uses: actions/checkout@master
+        with:
+          ref: ${{github.ref}}
+
+      # create or connect to ws new workspace
+      - name: Create Workspace
+        id: ws_create
+        uses: Azure/aml-workspace@v1
+        with:
+          azure_credentials: ${{ secrets.AZURE_CREDENTIALS }}
+          parameters_file: "test/test_workspace.json"
       
-      - name: Self test
-        id: selftest
-        uses: Azure/aml-run@master
+      # create or connect to compute
+      - name: Create Compute
+        id: compute
+        uses: Azure/aml-compute@v1
+        with:
+          azure_credentials: ${{ secrets.AZURE_CREDENTIALS }}
+          parameters_file: "test/test_aml_compute.json"
+
+      # send for training and do not wait.
+      - name: Train Script
+        id: ml_train
+        uses: ./
+        with:
+          azure_credentials: ${{ secrets.AZURE_CREDENTIALS }}
+          parameters_file: "test/train_withYamlRunConfig.json"
+
+      # send for training and do not wait.
+      - name: Train Script 2
+        id: ml_train2
+        uses: ./
         with:
           azure_credentials: ${{ secrets.AZURE_CREDENTIALS }}
-        continue-on-error: true
+          parameters_file: "test/train_withConfigpy.json"
+        
+      # No resource Clean-up is required in this case as resource provision and de-provision are not responsibilities of this action
```

Comment:
If we do not clean up in this action, will it be a separate action in this repo or somewhere else? Would it be easier to make a cleanup action?

Reply:
training is not a persistent things happening, What I meant was, we don't need to delete the workspace and compute after every training. 
we can have dedicated workspace and compute for running aml-train integration tests